CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 # Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-cmake_minimum_required(VERSION 3.8)
+cmake_minimum_required(VERSION 3.12)
```

Comment:
I'm not sure if this alright to upgrade, the internal RHEL5 Cmake version is capped at 3.9

Reply:
ah, that's a bummer. turns out the `CONFIGURE_DEPENDS` feature was introduced in 3.12, and it's a nice-to-have because we won't need to `clean` after adding files. from [the docs](https://cmake.org/cmake/help/latest/command/file.html#glob):

> New in version 3.12: If the CONFIGURE_DEPENDS flag is specified, CMake will add logic to the main build system check target to rerun the flagged GLOB commands at build time. If any of the outputs change, CMake will regenerate the build system.

i think we can keep the min version as-is and condition on version in the `file(GLOB ... )` call.

Reply:
No worries, It's a huge pain.. Main reason seems to be Cmake versions after 3.10 require C++11 which RHEL5 doesn't have. The if else statement at Line 11-29 is also there to get around some non-existent features in older versions of cmake.